### PR TITLE
Bridge Pangolin Parachain Alpha

### PR DESCRIPTION
**Pull request by jiguantong**
depends on
- [x] https://github.com/darwinia-network/darwinia-messages-substrate/pull/190

### CODE REVIEW

File: node/runtime/pangolin/src/lib.rs [hunk 1]
```diff
@@ -247,6 +247,13 @@ frame_support::construct_runtime! {
 		PangolinParachainFeeMarket: pallet_fee_market::<Instance2>::{Pallet, Call, Storage, Event<T>} = 64,
 		TransactionPause: module_transaction_pause::{Pallet, Call, Storage, Event<T>} = 54,
 
+		// pangolin <> pangolin parachain alpha bridge
+		BridgeMoonbaseRelayGrandpa: pallet_bridge_grandpa::<Instance3>::{Pallet, Call, Storage} = 67,
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/lib.rs:251]: `67` conflicts with `Preimage: pallet_preimage::{Pallet, Call, Storage, Event<T>} = 67,`

Comment (jiguantong) [reply node/runtime/pangolin/src/lib.rs:251]: Is `Preimage` in another branch? I didn't find it

Comment (AurevoirXavier) [reply node/runtime/pangolin/src/lib.rs:251]: Line 214.

Comment (AsceticBear) [reply node/runtime/pangolin/src/lib.rs:251]: image

Comment (jiguantong) [reply node/runtime/pangolin/src/lib.rs:251]: oh I see

File: node/runtime/pangolin/src/pallets/bridge_messages.rs [hunk 2]
```diff
@@ -77,3 +81,31 @@ impl Config<WithPangolinParachainMessages> for Runtime {
...
+	type InboundMessageFee = bp_pangolin_parachain::Balance;
+	type InboundPayload = bm_pangolin_parachain_alpha::FromPangolinParachainAlphaMessagePayload;
+	type InboundRelayer = bp_pangolin_parachain::AccountId;
+	type LaneMessageVerifier = bm_pangolin_parachain_alpha::ToPangolinParachainAlphaMessageVerifier;
+	type MaxMessagesToPruneAtOnce = MaxMessagesToPruneAtOnce;
+	type MaxUnconfirmedMessagesAtInboundLane = PangolinParachainMaxUnconfirmedMessagesAtInboundLane;
+	type MaxUnrewardedRelayerEntriesAtInboundLane =
+		PangolinParachainMaxUnrewardedRelayerEntriesAtInboundLane;
+	type MessageDeliveryAndDispatchPayment =
+		FeeMarketPayment<Self, WithPangolinParachainAlphaFeeMarket, Ring, RootAccountForPayments>;
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/pallets/bridge_messages.rs:97]: No `RootAccountForPayments` anymore, please check out the main branch.

File: node/runtime/pangolin/src/pallets/bridge_parachains.rs [hunk 3]
```diff
@@ -8,3 +10,8 @@ impl Config<WithRococoParachainsInstance> for Runtime {
 	type BridgesGrandpaPalletInstance = WithRococoGrandpa;
 	type HeadsToKeep = RococoHeadersToKeep;
 }
+impl Config<WithMoonbaseRelayParachainsInstance> for Runtime {
+	type BridgesGrandpaPalletInstance = WithMoonbaseRelayGrandpa;
+	type HeadsToKeep = RococoHeadersToKeep;
+	type ParasPalletName = PangolinParasPalletName;
```
Comment (AsceticBear) [inline node/runtime/pangolin/src/pallets/bridge_parachains.rs:16]: And this type

**Comment by jiguantong** [comment]
@AsceticBear  review again?

**Comment by AsceticBear** [comment]
Please fix CI first

**Review by AurevoirXavier — APPROVED** [review]

**Review by AsceticBear — APPROVED** [review]

**Pull request merged**